### ISSUE TITLE
Add a `skore.create_or_load`

### ISSUE THREAD

**Issue by Vincent-Maladiere** [probabl-ai/skore#917]
### Is your feature request related to a problem? Please describe.

When iterating on my modeling, I might often rerun the same notebook, sometimes dropping/removing entire intermediary or debug runs in skore when I don't need them (by deleting the project on my disk). So, I often switch between `skore.create()` and `skore.load()`.

### Describe the solution you'd like

Instead, I'd rather have a single `skore.create_or_load()` function that I could reuse all the time.

**Comment by thomass-dev** [comment]
The output of the `skore.create` function seems to be outdated. It returns the project.

**Comment by Vincent-Maladiere** [comment]
Ok, I haven't noticed the overwrite option. It's good to have a fresh start at every run during debugging phases, but for iterating on metrics I have to catch the exception as you showed.

I still think having a single entry point like `create_or_load()` would make sense, so that users like me don't need to bother with the exception catching.

**Comment by sylvaincom** [comment]
Or `project = skore.create([...])` just loads if the project already exists?

**Comment by thomass-dev** [comment]
> Or `project = skore.create([...])` just loads if the project already exists?

I disagree. It's important for user to explicitly know if he's working on an existing project, otherwise it could unknowingly overwrite important information.

The question is if we need `create_or_load()` or wheter letting the user `try/catch` is sufficient.

**Comment by MarieS-WiMLDS** [comment]
Asking the user to write try/except puts some load on them, and they are not software engineers. I'd prefer to find a solution where we have a way to do that straight with skore. I don't have an opinion between the other function and the other option for `existing` parameter. Both are quite straightforward and easy to understand.

**Comment by tuscland** [comment]
I like @augustebaum's suggestion.

And I agree with both of you:
- try/catch is not the target persona here. (though it would be acceptable for ML Engineers: it depends on the situation)
- We should avoid adding a new function.

**Comment by rouk1** [comment]
And what about merging create and load in a `open` function ?

It's signature could be:

```python
def open(project_path: str | Path="project.skore", create:bool=True):
    """Open a project given a project name or path.
    Parameters
        ----------
        project_path: str | Path
            The path where te project exists on disk.
        create: bool (defautl True)
            If set to false the function will raise an FileNotFoundError.
   """
```

**Comment by glemaitre** [comment]
I like @rouk1 proposal and more in favour of `create` name since it reflect to the `create` function in the API. An alternative is `autocreate=True/False` but I'm not sure it is better.

**Comment by augustebaum** [comment]
> And what about merging `create` and `load` in a `open` function ?

@rouk1 There should be an `overwrite` option as well but otherwise I think this is ready for dev

**Comment by thomass-dev** [comment]
The `create` and `overwrite` options should coexist.

**Comment by rouk1** [comment]
```python
def open(
    project_path: str | Path = "project.skore",
    *,
    auto_create: bool = True,
    overwrite: bool = False,
):
    """Open a project given a project name or path.

    Parameters
    ----------
    project_path: str | Path
        The path where te project exists on disk.
    auto_create: bool (default True)
        If ``True`` the function will create the projetc if it does not exists.
        If ``False`` the function will raise a FileNotFoundError if path dfoes not exist or is not a valid project.
    overwrite: bool (default True)
        If ``True``, overwrite an existing project with the same name.
        If ``False``, raise an error if a project with the same name already exists.


    Returns
    -------
    Project
        The corresponding Project instance.

    Raises
    ------
    FileNotFoundError
        Raised if:
        - path is not found and auto_create is set to ``False``
    ProjectCreationError
        Raised if:
        - project name is invalid
        - project already exists and overwrite is set to ``False``
    """
```

 WDYT ?

**Comment by augustebaum** [comment]
I'm implementing this and having second thoughts about the name `open`, since it conflicts with the standard library function of the same name. Any thoughts? @rouk1 @MarieS-WiMLDS

**Comment by rouk1** [comment]
🤔 

Pillow style ?

```python
from skore import Project

p = Project.open()
```

Let it go and encourage user to write ?

```python
import skore as sk

p = sk.open()
```

**Comment by MarieS-WiMLDS** [comment]
Why did we drop an option like `open_or_create`?  `access_project`, or `open_project`?  
Just dropping ideas.

**Comment by augustebaum** [comment]
What should happen if the user calls `open("project.skore", create=True, overwrite=False)` (the default), but `project.skore` already exists? Should we try to create a project (and thus fail), or should we load the existing project, which contradicts the `create` setting?

I'm starting to circle back to
```python
# Equivalent to current `overwrite=True`
project = skore.create("my_project", existing="overwrite")

# New behaviour (could be the default IMO)
project = skore.create("my_project", existing="load")
```

**Comment by augustebaum** [comment]
> Why did we drop an option like `open_or_create`? `access_project`, or `open_project`? Just dropping ideas.

I have a slight preference for one-word commands when possible, but these are all valid.

**Comment by thomass-dev** [comment]
what about `project = skore.load("my_project", default=<"raise"|"create"|"overwrite">)`

**Comment by rouk1** [comment]
> What should happen if the user calls `open("project.skore", create=True, overwrite=False)` (the default), but `project.skore` already exists? Should we try to create a project (and thus fail), or should we load the existing project, which contradicts the `create` setting?

To me you should just open the existing project. For a user open should return a project no matter if it exists already or not. Also we should not delete user data when calling open. Deleting must be a conscious action.

**Comment by augustebaum** [comment]
> Also we should not delete user data when calling open.

Are you saying the `overwrite` option should be removed?

**Comment by rouk1** [comment]
> > Also we should not delete user data when calling open.
> 
> Are you saying the `overwrite` option should be removed?

Maybe yes, at least it's default value should be false. (imo)

**Comment by MarieS-WiMLDS** [comment]
I really like the overwrite option for testing! Let's keep the option but indeed, set the default value to false.

**Comment by MarieS-WiMLDS** [comment]
From this issue https://github.com/probabl-ai/skore/issues/1000, I wonder why don't we do `skore.Project(name: "blabla", overwrite = True)`? It would be an init function. instead of having a separate one.

**Comment by GaelVaroquaux** [comment]
Related: https://github.com/probabl-ai/skore/issues/1002
(I created the above issue without having seen this one)

**Comment by Vincent-Maladiere** [comment]
So, what is the conclusion on this?

**Comment by thomass-dev** [comment]
You have now a new `open` function https://github.com/probabl-ai/skore/pull/992.

`my_project = skore.open("quick_start", overwrite=True)`
